ci/unit/test_main_node.sh

```diff
@@ -26,12 +26,9 @@ cd "$DAOS_BASE"
 if [ "$WITH_VALGRIND" = "memcheck" ]; then
     # run_test.sh with valgrind memcheck
     IS_CI=true OLD_CI=false RUN_TEST_VALGRIND=memcheck utils/run_test.sh
-    ls
-    ls test_results
     # Remove DAOS_BASE from memcheck xml results
-    if ls test_results/results-*-memcheck.xml; then
-        sed -i "s:$DAOS_BASE::g" test_results/results-*-memcheck.xml
-    fi
+    find test_results -maxdepth 1 -name 'results-*-memcheck.xml' \
+        -exec sed -i "s:$DAOS_BASE::g" {} +
```

Comment:
Not sure why there is a trailing `+` here, but that aside, it's usually more efficient to use `xargs` with `find` rather than `-exec`:

```suggestion
    find test_results -maxdepth 1 -name 'results-*-memcheck.xml' |
        xargs -r sed -i "s:$DAOS_BASE::g"
```

Reply:
I had something similar before, but I was having problems figuring out this linter error
`(lint) Use 'find .. -print0 | xargs -0 ..' or 'find .. -exec .. +' to allow non-alphanumeric filenames. [SC2011]`
but I think I can make it work with -print0 and xargs -0, I'll fix it on my next patch

Reply:
Yes, `-print0 | xargs -0` is the solution.